src/lib/web3-utils.js

```diff
@@ -148,3 +159,41 @@ export function transformAddresses(str, callback) {
       callback(part, ETH_ADDRESS_TEST_REGEX.test(part), index)
     )
 }
+
+/**
+ * Check if the ETH node at the given URI is compatible for the current environment
+ * @param {string} uri URI of the ETH node.
+ * @param {string} expectedNetworkType The expected network type of the ETH node.
+ * @returns {Promise} Resolves if the ETH node is compatible, otherwise throws:
+ *    - InvalidURI: URI given is not compatible (e.g. must be WebSockets)
+ *    - InvalidNetworkType: ETH node connected to wrong network
+ *    - NoConnection: Couldn't connect to URI
+ */
+export async function checkValidEthNode(uri) {
+  // Must be websocket connection
+  const isLocalOrUnknown = isLocalOrUnknownNetwork(env('CHAIN_ID'))
+  if (!isLocalOrUnknown) {
+    if (!websocketRegex.test(uri)) {
```

Comment:
I think we should do this check regardless of the network being local or unknown—we only support http/https right now.